build.gradle

```diff
@@ -313,3 +313,46 @@ task ktlintFormat(type: JavaExec, group: "formatting") {
 compileKotlin { kotlinOptions.freeCompilerArgs = ['-Xjsr305=strict'] }
 
 apply from: 'build-tools/pkgbuild.gradle'
+
+// This IT is to simulate the situation
+// when there are old version (without metadata change)
+// and new version mixed in one cluster
+import org.elasticsearch.gradle.test.RestIntegTestTask
+
+def mixedClusterTest = project.tasks.create('mixedCluster', RestIntegTestTask.class)
+def mixedClusterFlag = findProperty('mixed') as Boolean ?: false
+println("mixed cluster flag: $mixedClusterFlag")
+mixedClusterTest.dependsOn(bundlePlugin)
+
+testClusters.mixedCluster {
+    testDistribution = "OSS"
+    if (_numNodes > 1) numberOfNodes = _numNodes
+    getNodes().each { node ->
+        node.plugin(provider({
+            new RegularFile() {
+                @Override
+                File getAsFile() { fileTree("src/test/resources/job-scheduler").getSingleFile() }
+            }
+        }))
+
+        if (mixedClusterFlag && node.name == "mixedCluster-1") {
+            node.plugin(provider({
+                new RegularFile() {
+                    @Override
+                    File getAsFile() { fileTree("src/test/resources/index-management").getSingleFile() }
```

Comment:
If this is supposed to contain an older version without the metadata change, should we maybe rename the file uploaded to `src/test/resources/index-management` to make that clearer?

Also, as time goes on, what happens if this plugin version gets very old. Do we hit errors if it is a major version apart for the ES version it's built for compared to the branch running the test?

Reply:
This is only for one regression integration test. To simulate we have 2 versions of ISM plugin
It requires the `old` ISM plugin to use same ES version, but different plugin version, like this time 1.13.0.0, and 1.13.0.1 for `new` plugin.
I am not sure how to manage this test in the future ES version though... since if we comes to a new version, then we already has this change, we don't have an `old` plugin. 
And this testing framework, I am not sure it even allows 2 nodes running different ES version, prob not.

Reply:
Wonder if there's a way we can inject custom metadata into the cluster state and create older jobs to mimic old version of ISM existing. Wouldn't be able to do it from the RestTests I'd imagine, but maybe the other integ test that gives more internal control..

Reply:
I am using the Integ Test, to inject custom metadata to simulate the situation. Can check the test in `MetadataRegressionIT`

Reply:
Ah cool just haven't gotten to it yet :)